asv_bench/benchmarks/io/excel.py

```diff
@@ -47,6 +47,25 @@ def time_write_excel(self, engine):
         writer.save()
 
 
+class WriteExcelStyled:
+    params = ["openpyxl", "xlsxwriter"]
+    param_names = ["engine"]
+
+    def setup(self, engine):
+        self.df = _generate_dataframe()
+
+    def time_write_excel_style(self, engine):
```

Comment:
I'm not sure I follow. The benchmark should include cache hits because the cached function is run for each cell, so the example should have 3 misses and `(N*C - 3)` hits. I added a test in [8600ba6](https://github.com/pandas-dev/pandas/pull/47371/commits/8600ba641956eab195cd305ffab7cb4ceb208b09) that ensures the cache is actually being hit.